traits/ctraits.c

```diff
@@ -4253,6 +4253,32 @@ _trait_comparison_mode(trait_object *trait, PyObject *args)
     return Py_None;
 }
 
+/*-----------------------------------------------------------------------------
+|  getter for trait comparison mode
++----------------------------------------------------------------------------*/
+
+static PyObject *
+_get_trait_comparison_mode_int(trait_object *trait, PyObject *Py_UNUSED(ignored))
+{
+    PyLongObject *comparison_mode = PyLong_FromLong(2);
```

Comment:
If either of the `if` branches below gets executed, then we never check for an error return in this branch. (We end up calling `PyLong_FromLong` twice; the second call gets checked, but not the first.)

I'd suggest instead reworking so that `PyLong_FromLong` only gets called once. I'd probably go with an extra variable of type `int`, e.g.:
```
int i_comparison_mode;
if (trait->flags & TRAIT_NO_VALUE_TEST) {
    i_comparison_mode = 0;
}
else if (...) {
    i_comparison_mode = 1;
}
else {
    i_comparison_mode = 2;
}
comparison_mode = PyLong_FromLong(i_comparison_mode);
...
```